internal/npm_install/generate_build_file.js

```diff
@@ -672,53 +660,82 @@ function filterFilesForFilegroup(files, exts = []) {
 }
 
 /**
- * Given a `pkg`, return the skylark `filegroup` target which is the default
- * target for the alias `@npm//${pkg.name}`.
+ * Returns true if the specified `pkg` conforms to Angular Package Format (APF),
+ * false otherwise. If the package contains `*.metadata.json` and a
+ * corresponding sibling `.d.ts` file, then the package is considered to be APF.
  */
-function printPkgTarget(pkg, pkgDeps) {
-  return `
-filegroup(
-    name = "${pkg._name}__pkg",
-    srcs = [
-        # ${pkg._dir} package contents (and contents of nested node_modules)
-        ":${pkg._name}__files",
-        # direct or transitive dependencies hoisted to root by the package manager
-        ${
-      pkgDeps.map(dep => `"//node_modules/${dep._dir}:${dep._name}__files",`).join('\n        ')}
-    ],
-    tags = ["NODE_MODULE_MARKER"],
-)
-`;
+function isNgApfPackage(pkg) {
+  const set = new Set(pkg._files);
+  const metadataExt = /\.metadata\.json$/;
+  return pkg._files.some((file) => {
+    if (metadataExt.test(file)) {
+      const sibling = file.replace(metadataExt, '.d.ts');
+      if (set.has(sibling)) {
+        return true;
+      }
+    }
+    return false;
+  });
+}
+
+/**
+ * If the package is in the Angular package format returns list
+ * of package files that end with `.umd.js`, `.ngfactory.js` and `.ngsummary.js`.
+ */
+function getNgApfScripts(pkg) {
+  return isNgApfPackage(pkg) ?
+      filterFiles(pkg._files, ['.umd.js', '.ngfactory.js', '.ngsummary.js']) :
+      [];
 }
 
 /**
- * Given a pkg, return the skylark `filegroup` and/or `ng_apf_library` targets for the package.
+ * Given a pkg, return the skylark `node_module_library` targets for the package.
  */
 function printPackage(pkg) {
-  const sources = filterFilesForFilegroup(pkg._files, INCLUDED_FILES);
-  const dtsSources = filterFilesForFilegroup(pkg._files, ['.d.ts']);
+  const sources = filterFiles(pkg._files, INCLUDED_FILES);
+  const dtsSources = filterFiles(pkg._files, ['.d.ts']);
+  // TODO(gmagolan): add UMD & AMD scripts to scripts even if not an APF package _but_ only if they
```

Comment:
We can probably add `.umd.js` files from all npm packages to `scripts` but we just have to detect if they are named-UMD like `node_modules/rxjs/bundles/rxjs.umd.js`. Unnamed UMD will break downstream rules.